cpp/include/raft/neighbors/ivf_pq_types.hpp

```diff
@@ -104,6 +104,25 @@ struct index_params : ann::index_params {
    * flag to `true` if you prefer to use as little GPU memory for the database as possible.
    */
   bool conservative_memory_allocation = false;
+
+  /**
+   * Helper that sets values according to the extents of the dataset mdspan.
+   */
+  template <typename DataT, typename Accessor>
+  static index_params from_dataset(
+    mdspan<const DataT, matrix_extent<int64_t>, row_major, Accessor> dataset,
+    raft::distance::DistanceType metric = raft::distance::L2Expanded)
+  {
+    index_params params;
+    params.n_lists =
+      dataset.extent(0) < 4 * 2500 ? 4 : static_cast<uint32_t>(std::sqrt(dataset.extent(0)));
+    params.pq_dim =
+      round_up_safe(static_cast<uint32_t>(dataset.extent(1) / 4), static_cast<uint32_t>(8));
+    params.pq_bits                  = 8;
+    params.kmeans_trainset_fraction = dataset.extent(0) < 10000 ? 1 : 0.1;
+    params.metric                   = metric;
+    return params;
+  }
```

Comment:
These are specific default params that are useful for constructing a CAGRA index, but possibly not ideal as default params for an  `ivf_pq` index. The documentation has to make this clear. I think it would be also preferred to keep this helper in the CAGRA namespace.

Reply:
Actually, I don't know that I agree with this. The sqrt(n) and compression ratio are reasonable defaults for ivfpq in general, irrespective of CAGRA.

Reply:
Ok, maybe you are right. It is specific for CAGRA only in the sense that it leads to good enough recall for a building a CAGRA graph. But we do not give any recall guarantees (we cannot since it is dataset dependent) so we can treat this general defaults.